src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -356,11 +360,17 @@ public async Task SaveChangesAsync(IEnumerable<UnspentOutputs> unspentOutputs, I
                 }
 
                 this.blockHash = nextBlockHash;
+                var rewindData = new RewindData(nextBlockHash);
+
                 foreach (UnspentOutputs unspent in unspentOutputs)
                 {
-                    CacheItem existing;
-                    if (this.unspents.TryGetValue(unspent.TransactionId, out existing))
+                    if (this.unspents.TryGetValue(unspent.TransactionId, out CacheItem existing))
                     {
+                        // We'll need to restore the original outputs.
+                        UnspentOutputs clone = unspent.Clone();
+                        clone.Outputs = existing.UnspentOutputs?.Outputs.ToArray() ?? Array.Empty<TxOut>();
```

Comment:
Is this correct? I'm not sure you should be using `existing` here is that seems to be an old version of the structure. Should you not use `unspent`? The previous line made a full copy of `unspent`and my question is why would we want to change anything about that. The outputs were copied and now you're replacing that with something else. I don't really understand why. If you look down, you can see that we do even modify this `existing` thing, which would suggest that whatever you have on this line cannot be correct. But I'm not sure